carbonmark/components/pages/Project/Retire/Pool/RetireForm.tsx

```diff
@@ -126,12 +132,17 @@ export const RetireForm: FC<Props> = (props) => {
     isProcessing;
 
   const showTransactionView = !!inputValues && !!allowanceValue;
-  const disableSubmit =
-    !quantity ||
-    Number(quantity) <= 0 ||
-    (paymentMethod === "fiat"
-      ? Number(costs) < Number(fiatMinimum)
-      : Number(quantity) < Number(MINIMUM_TONNE_QUANTITY));
+
+  const disableSubmit = () => {
+    if (!quantity || Number(quantity) <= 0) return true;
+    if (paymentMethod === "fiat") {
+      return Number(costs) < Number(fiatMinimum);
+    } else if (paymentMethod === "bank-transfer") {
+      return Number(quantity) < Number(MINIMUM_TONNE_QUANTITY_BANK_TRANSFER);
+    } else {
+      return Number(quantity) < Number(MINIMUM_TONNE_QUANTITY);
+    }
+  };
```

Comment:
Does this need to be a function? quantity and paymentMethod are watched show it should re-render the component when they change

Reply:
@0xemc valid point, I initially changed this as I found the ternary was getting complicated to read so decided to try clean it up and make it a bit more readable. Let me take another look at this though.